src/gaen/nativeModule.ts

```diff
@@ -142,6 +142,16 @@ export const isBluetoothEnabled = async (): Promise<boolean> => {
   return bluetoothStatus === true || bluetoothStatus === "true"
 }
 
+export const doesDeviceSupportLocationlessScanning = async (): Promise<
+  boolean
+> => {
+  return deviceInfoModule.deviceSupportsLocationlessScanning()
```

Comment:
I'm probably just used to Elm, but why don't we have to account for the fact that the promise could reject here? E.g.

```
try { return deviceInfoModule.deviceSupportsLocationlessScanning() } catch { ... }
```

Reply:
This is a pretty deep question, so perhaps worth chatting about irl, and theres a few reasons worth pointing out here so I'll just mention that `deviceInfoModule.deviceSupportsLocationlessScanning()` is of type `any` so the compiler has no way of understanding what this returns.  you can play around with updating the type signature to anything else and notice that the complier wont complain at all.

You can kinda think of these functions in this file as wrappers around the IO edge of our system which doesnt have any type information inherently. So here we are doing the work of provide the basic type the compiler will check against for the rest of our system. Which is why type bugs are likely to be common in this file and are worth taking and extra look at and providing integration tests for.

Reply:
Ah I see. Very interesting. Thanks for the thorough explanation!